example/example_test.go

```diff
@@ -85,10 +85,22 @@ var cases = []struct {
 			"page_number": 1,
 			"result_per_page": 10]
 		}`,
-		ExpectedStatusCode:  422,
+		ExpectedStatusCode:  400,
 		ExpectedJSONResBody: ``,
 	},
 	{
+		Name: "test response nil",
+		URL:  "/example.SearchService/SearchReturnNil",
+		JSONReqBody: `{
+			"query": "query string",
+			"page_number": 1,
+			"result_per_page": 10
+		}`,
+		ExpectedStatusCode:  200,
+		ExpectedJSONResBody: ``,
```

Comment:
You can see the problem clearly with a JSON response:

1. Client makes request

2. Server responds with `200 OK`

3. Response body is `` (empty string, `Content-Length: 0`)

4. Client says "200? Great! let's feed the body into `JSON.parse`:

   ```
   JSON.parse("")
   VM195668:1 Uncaught SyntaxError: Unexpected end of JSON input
       at JSON.parse (<anonymous>)
       at <anonymous>:1:6
   ```

---

> return Empty,still with 204?

If you return 204, there is *no* response body. This is what 204 is defined to mean, "No Content":

> A 204 response is terminated by the first empty line after the header fields because it cannot contain a message body.

https://tools.ietf.org/html/rfc7231#section-6.3.5

`Empty` is a valid protobuf message, so you are returning content. But if the service doesn't define that it returns `Empty`, then prottp should not return `Empty`? It should return exactly what was defined in the `.proto` file, or a 4xx/5xx error.

---

> What about those errors like 400 Bad request

If the response code is 400, then the body can be whatever or empty. If you have some content in the body, it would be nice (but not necessary) for the `Content-Type` to match.

Here's what the spec has to say about 4xx bodies:

> Except when responding to a HEAD request, the server SHOULD send a representation containing an explanation of the error situation

So you could return a message saying "you had an error at byte X in your JSON` or something like that...